index.bs

```diff
@@ -1206,8 +1206,27 @@ This specification defines <dfn>context cleanup steps</dfn> as the following ste
 
 ## Garbage Collection ## {#web-transport-gc}
 
-The user agent MUST NOT garbage collect a {{WebTransport}} object whose {{[[State]]}} is either
-`"connecting"` or `"connected"`.
+A {{WebTransport}} object whose {{[[State]]}} is `"connecting"` must not be garbage collected if
+{{[[IncomingBidirectionalStreams]]}}, {{[[IncomingUnidirectionalStreams]]}}, any
+{{WebTransportReceiveStream}}, or {{[[Datagrams]]}}.{{WebTransportDatagramDuplexStream/[[Readable]]}}
+is [=ReadableStream/locked=], or if the {{ready}}, {{draining}}, or {{closed}} promise is being observed.
+
+A {{WebTransport}} object whose {{[[State]]}} is `"connected"` must not be garbage collected if
+{{[[IncomingBidirectionalStreams]]}}, {{[[IncomingUnidirectionalStreams]]}}, any
+{{WebTransportReceiveStream}}, or {{[[Datagrams]]}}.{{WebTransportDatagramDuplexStream/[[Readable]]}}
+is [=ReadableStream/locked=], or if the {{draining}} or {{closed}} promise is being observed.
+
+A {{WebTransport}} object whose {{[[State]]}} is `"draining"` must not be garbage
+collected if the {{closed}} promise is being observed.
```

Comment:
Hmm, on second thought, can we really GC here if readables are locked? While "draining" is a loose promise from the server not to generate new streams, there might still be data in receive buffers, so I worry this is a race. WDYT?

Reply:
That's a good point, thank you! I added the part about locked readables here as well.